core/htmlparser/filter.js

```diff
@@ -402,6 +402,140 @@
 } )();
 
 /**
+ * Abstract class describing the definition of {@link CKEDITOR.htmlParser.filter} rules.
+ *
+ * This class illustrates the properties that developers can use to define and create filter rules.
+ *
+ * A filter rules definition object represents rules as a set of properties with callback functions
+ * to be applied for transforming and filtering content upon data processing.
+ *
+ * Filter rules definition can be also added with {@link CKEDITOR.htmlParser.filter.addrules}.
+ *
  * @class CKEDITOR.htmlParser.filterRulesDefinition
  * @abstract
  */
+
+/**
+ * @property {String[][]/RegExp[][]} elementNames An array of rules for data processing element names.
+ * Rule is defined as an array with two items [ string/regex, string ].
+ * Every matching string from first item will be converted into second.
+ *
+ * Examples:
+ *
+ * ```javascript
+ * elementNames: [
+ * 		[ /^div$/, 'p' ], // Converts 'div' into 'p'.
+ * 		[ /^cke:?/, '' ] // Removes 'cke:' prefixes.
+ * ]
+ * ```
+ *
+ * Incorrect:
+ *
+ * ```javascript
+ * elementNames: [
+ * 		[ 'p', 'div' ] // Converts every letter 'p' inside element name into 'div'
+ * 		// 'p' element will be converted into 'div', 'span' wil be converted into 'sdivan'.
+ * 		// To convert only 'p' elements use regex `/^p$/`.
+ * ]
+ * ```
+ *
+ */
+
+/**
+ * @property {String[][]/RegExp[][]} attributeNames An array of rules for data processing attribute names.
+ * Rule is defined as an array with two items [ string/regex, string ].
+ * Every matching string from first item will be converted into second.
+ *
+ * Examples:
+ *
+ * ```javascript
+ * attributeNames: [
+ * 		[ 'data-foo', 'data-bar' ],
+ * 		// Converts string in attribute name from 'data-foo' into 'data-bar'
+ * 		// Note that attribute 'data-foo-baz' is converted into 'data-bar-baz'.
+ *
+ * 		[ /^data-custom$/, 'data-cke' ]
+ * 		// Converts attribute 'data-custom' into 'data-cke'.
+ * ]
+ * ```
+ *
+ */
+
+/**
+ * @property {Object.<String, Function>} elements An object containing pairs of keys representing
+ * element selectors and function for data processing selected elements. If function contains return statement element will be removed.
```

Comment:
I would split this long comment into two lines (after the dot).

Also, sencence `If function contains return statement element will be removed` is invalid. Every javascript function returns something (also without a return statement, which is `undefined`) so you should be more precise about return type. What if I write:
```javascript
function() {
  return undefined;
}
```
Will element still be removed?

Reply:
This `data processing` phrase is very mysterious to me. I really don't understand what it does and what my function should do. Something like `transformation function` is more understable for me than it, although it may be subjective. E.g.

`An object containing pairs of keys representing element selectors and transformation function.`

Also, is the element really selected inside the editor? What if I apply filter into document fragment?